torch/nn/parallel/scatter_gather.py

```diff
@@ -39,6 +39,11 @@ def scatter_kwargs(inputs, kwargs, target_gpus, dim=0):
         inputs.extend([() for _ in range(len(kwargs) - len(inputs))])
     elif len(kwargs) < len(inputs):
         kwargs.extend([{} for _ in range(len(inputs) - len(kwargs))])
+
+    if len(inputs) == 0 and len(kwargs) == 0:
```

Comment:
Should these two lines be consolidated into the else clause in line 36-37 above?

Reply:
Yes, and if I don't force them to match the number of target GPUs, the `split()` in the test case isn't required.